### PR TITLE
LPS-157269 | Can get multiple ManyToMany relationship details as nested fields

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Add a test to assert can get multiple ManyToMany relationship details as nested fields

**Test Plan**
Given university created
And Given subject created
And Given student
And Given manyToMany relationship universitySubjects created
And Given manyToMany relationship universityStudents created
And Given university entry created
And Given subject entry created
And Given student entry created
And Given in c/subjects I use PUT /{subjectId}/universitiesSubjects/{universityId}
And Given in c/students I use PUT /{studentId}/universitiesStudents/{universityId}
When I call /o/c/universities?nestedFields=universitiesSubjects,universitiesStudents
Then I receive correct information about the related objects

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-157269

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase [hunk 1]
```diff
@@ -553,6 +555,123 @@ definition {
...
+				deletionType = "cascade",
+				en_US_label = "universitiesSubjects",
+				name = "universitiesSubjects",
+				objectDefinitionId1 = "${objectDefinitionId1}",
+				objectDefinitionId2 = "${objectDefinitionId2}",
+				type = "manyToMany");
+		}
+
+		task ("Given manyToMany relationship universityStudents created") {
+			var objectDefinitionId1 = JSONObject.getObjectId(objectName = "University");
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:584]: I believe declaring the University here is redundant, the line 571 should be enough.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase [hunk 2]
```diff
@@ -553,6 +555,123 @@ definition {
...
+
+			ObjectDefinitionAPI.assertNestedFieldDetail(
+				expectedValue = "Liferay Employee",
+				getObjectsWithNestedFieldReponse = "${getObjectsWithNestedFieldReponse2}",
+				nestedField = "universitiesStudents",
+				objectEntryId = "${universityId}",
+				objectField = "name");
+		}
+	}
+
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:674]: I don't see some test steps declared in the test. Are you sure you're automating the correct one?


```
Test Plan
Given university created
And Given subject created
And Given student
And Given manyToMany relationship universitySubjects created
And Given manyToMany relationship universityStudents created
And Given university entry created
And Given subject entry created
And Given student entry created
And Given in c/subjects I use PUT /{subjectId}/universitiesSubjects/{universityId}
And Given in c/students I use PUT /{studentId}/universitiesStudents/{universityId}
When I call /o/c/universities?nestedFields=universitiesSubjects,universitiesStudents
Then I receive correct information about the related objects
```

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase [hunk 3]
```diff
@@ -553,6 +555,104 @@ definition {
...
+		task ("Given I relate student entry with university entry through students PUT endpoint") {
+			ObjectDefinitionAPI.relateObjectEntries(
+				en_US_plural_label = "students",
+				objectEntry1 = "${studentId}",
+				objectEntry2 = "${universityId}",
+				relationshipName = "universitiesStudents");
+		}
+
+		task ("When I call universities GET endpoint with nestedFields=universitiesSubjects,universitiesStudents") {
+			var getObjectsWithNestedFieldReponse1 = ObjectDefinitionAPI.getObjectsWithNestedField(
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:630]: this should be one call, not two. 
```
ObjectDefinitionAPI.getObjectsWithNestedField(
				nestedField = "universitiesStudents,universitiesSubjects",
				objects = "universities");
```

**Review by magjed4289 — CHANGES_REQUESTED** [review]

**Review by magjed4289 — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**